src/SqlFormatter.php

```diff
@@ -748,144 +748,144 @@ final class SqlFormatter
 
     /** @var string */
-    public static $cliWord = '';
+    public $cliWord = '';
 
     /** @var string */
-    public static $cliError = "\x1b[31;1;7m";
+    public $cliError = "\x1b[31;1;7m";
 
     /** @var string */
-    public static $cliComment = "\x1b[30;1m";
+    public $cliComment = "\x1b[30;1m";
 
     /** @var string */
-    public static $cliFunctions = "\x1b[37m";
+    public $cliFunctions = "\x1b[37m";
 
     /** @var string */
-    public static $cliVariable = "\x1b[36;1m";
+    public $cliVariable = "\x1b[36;1m";
 
     /**
      * The tab character to use when formatting SQL
      *
      * @var string
      */
-    public static $tab = '  ';
+    public $tab = '  ';
 
     /**
      * This flag tells us if queries need to be enclosed in <pre> tags
      *
      * @var bool
      */
-    public static $usePre = true;
+    public $usePre = true;
 
     /**
      * This flag tells us if SqlFormatted has been initialized
      *
      * @var bool
      */
-    private static $init;
+    private $init;
 
     // Regular expressions for tokenizing
 
     /** @var string */
-    private static $regexBoundaries;
+    private $regexBoundaries;
 
     /** @var string */
-    private static $regexReserved;
+    private $regexReserved;
 
     /** @var string */
-    private static $regexReservedNewline;
+    private $regexReservedNewline;
 
     /** @var string */
-    private static $regexReservedToplevel;
+    private $regexReservedToplevel;
 
     /** @var string */
-    private static $regexFunction;
+    private $regexFunction;
 
     // Cache variables
     // Only tokens shorter than this size will be cached.  Somewhere between 10
     // and 20 seems to work well for most cases.
     /** @var int */
-    public static $maxCachekeySize = 15;
+    public $maxCachekeySize = 15;
 
     /** @var mixed[][] */
-    private static $tokenCache = [];
+    private $tokenCache = [];
 
     /** @var int */
-    private static $cacheHits = 0;
+    private $cacheHits = 0;
 
     /** @var int */
-    private static $cacheMisses = 0;
+    private $cacheMisses = 0;
 
     /**
      * Get stats about the token cache
      *
      * @return mixed[] An array containing the keys 'hits', 'misses', 'entries', and 'size' in bytes
      */
-    public static function getCacheStats() : array
+    public function getCacheStats() : array
     {
         return [
-            'hits'=>self::$cacheHits,
-            'misses'=>self::$cacheMisses,
-            'entries'=>count(self::$tokenCache),
-            'size'=>strlen(serialize(self::$tokenCache)),
+            'hits'=>$this->cacheHits,
+            'misses'=>$this->cacheMisses,
+            'entries'=>count($this->tokenCache),
+            'size'=>strlen(serialize($this->tokenCache)),
         ];
     }
 
     /**
      * Stuff that only needs to be done once.  Builds regular expressions and sorts the reserved words.
      */
-    private static function init() : void
+    private function init() : void
```

Comment:
I hope that getting rid of all this `static` does not hurt performance. I see that this method uses was meant to cache some regex expressions

Reply:
I think it won't if people are careful to reuse the same instance, which is the case nowadays with DI.